plane/src/types/backend_state.rs

```diff
@@ -273,50 +304,53 @@ impl BackendState {
         BackendState::Ready { address }
     }
 
-    pub fn to_terminating(
-        &self,
-        termination: TerminationKind,
-        reason: TerminationReason,
-    ) -> BackendState {
+    pub fn to_hard_terminating(&self, reason: TerminationReason) -> BackendState {
+        if self.status() >= BackendStatus::HardTerminating {
+            tracing::warn!(?reason, state=?self, "to_hard_terminating called on backend in later state.");
+            return self.clone();
+        }
+
         match self {
-            BackendState::Terminated { .. } => {
-                tracing::warn!(?reason, termination=termination.as_value(), state=?self, "to_terminating called on terminated backend");
-                self.clone()
-            }
-            // a soft terminating backend can be transitioned to a hard terminating backend
-            BackendState::Terminating {
-                last_status,
-                termination: termination_kind,
-                ..
-            } => {
-                if termination_kind == &termination {
-                    tracing::warn!(?reason, termination=termination.as_value(), state=?self, "to_terminating called on terminating backend with the same termination_kind");
-                    return self.clone();
-                }
-                if termination_kind != &TerminationKind::Hard {
-                    tracing::warn!(?reason, termination=termination.as_value(), state=?self, "to_terminating called on terminating backend with soft termination kind");
-                    return self.clone();
-                }
-                BackendState::Terminating {
-                    last_status: *last_status,
-                    termination,
-                    reason,
-                }
-            }
-            _ => BackendState::Terminating {
+            BackendState::Terminating { last_status, .. } => BackendState::HardTerminating {
+                last_status: *last_status,
+                reason,
+            },
+            _ => BackendState::HardTerminating {
                 last_status: self.status(),
-                termination,
                 reason,
             },
         }
     }
 
+    pub fn to_terminating(&self, reason: TerminationReason) -> BackendState {
```

Comment:
Nit: maybe put `to_terminating` before `to_hard_terminating` so all these status methods are in order